Backbone.API/Extensions/IServiceCollectionExtensions.cs

```diff
@@ -114,6 +137,14 @@ public static class IServiceCollectionExtensions
         return services;
     }
 
+    public static object GetPropertyValue(object source, string propertyName)
+    {
+        foreach (var property in propertyName.Split('.').Select(s => source.GetType().GetProperty(s)))
+            source = property.GetValue(source, null);
```

Comment:
Isn't this the same as 

```cs
var source = propertyName.Split('.').Select(n => source.GetType().GetProperty(n)).Select(p => p.GetValue(source, null))
```
?

A little bit of formatting to make it more readable:

```cs
var source = propertyName
                    .Split('.')
                    .Select(n => source.GetType().GetProperty(n))
                    .Select(p => p.GetValue(source, null))
```

And you should probably rename `propertyName` to `propertyPath`

Reply:
Nvm, that doesn't make any sense. ;-)